### DISCUSSION TITLE
Problem with row selection

### DISCUSSION THREAD

**Discussion by dlauterbach** [TanStack/table#2431] [General] [answered]
I'm attempting to use row selection but I'm experiencing problems with the checkboxes.  The first selected checkbox appears to work fine, but subsequent checkbox selects are not working.  Subsequent checkboxes do not show as selected and do not update the diagnostic block on the page and there are no errors in the dev tools console.

To isolate the problem I copied App.js from the row selection example in this repo over App.js in my project.  The same issue occurred.  Then I cloned this repo, ran "npm install", and started a new development server.  It worked!  Same (App.js) code in different npm environments behaves differently.  Any ideas how I can isolate which npm module is causing my problem?  My project is new and was bootstrapped with CRA.

**Comment by 07harish** [comment]
Hey, 
Check if you are doing setState somewhere, Table might reset your selected rows & you will have to write code to skip reset, Check faq's in the docs (autoResetSelectedRows: !skipPageResetRef.current)

Share codesandbox if possible.

If you doing manual pagination? If so, then there is a different approach to this.

**Reply by dlauterbach** [reply]
I copied the row-selection example code (App.js, makeData.js) into a brand new CRA project and it still does not work.  I'm not making any changes to the code.  No additional setting of state or pagination.  I don't think the docs will help as I cannot get this code from the repo to work in a bare bones CRA project.  I even deleted node_modules in my new CRA project and copied package.json and package_lock.json from row-selection example and then re-ran "npm install".  I then used "npm list <package_name>" to compare package versions between my new CRA project and my clone of row-selection.  All the versions are the same, yet the checkboxes work in the cloned project but fail in my new CRA project.  What am I missing?

**Comment by dlauterbach** [comment, marked as answer]
I figure it out!  CRA included "React.StrictMode" tags around my application root component in index.js.  By removing these tags my checkboxes are now working as they should.  @tannerlinsley - Is this a known issue?  Sorry I'm new to React.

**Reply by tannerlinsley** [reply]
RT should be strict model compliant, so yes, this is unexpected and likely a bug.
